### PR TITLE
Add support for parameter-less arrow expressions

### PR DESCRIPTION
**Pull request by Anoukh**
### Purpose
This PR will support writing arrow expressions without an input parameter
```ballerina
function () returns string lambda = () => "Some Text";
```

### CODE REVIEW

File: tests/ballerina-unit-test/src/test/resources/test-src/expressions/lambda/arrow-expression.bal [hunk 1]
```diff
@@ -175,3 +175,15 @@ function testArrowExprInObject() returns int {
...
+
+function testArrowExprWithNoArguments() returns string {
+    function () returns string lambda = () => "Some Text " + logString;
+    return lambda();
+}
+
+function testArrowExprWithNoArgumentsAndStrTemplate() returns string {
+    function () returns string lambda = () => string`Some Text {{logString}}`;
+    return lambda();
+}
```
Comment (Kishanthan) [inline tests/ballerina-unit-test/src/test/resources/test-src/expressions/lambda/arrow-expression.bal]: Need test cases for closures as well with noArg arrow expressions.

Comment (Anoukh) [reply tests/ballerina-unit-test/src/test/resources/test-src/expressions/lambda/arrow-expression.bal]: `logString` is a package level variable

Comment (Kishanthan) [reply tests/ballerina-unit-test/src/test/resources/test-src/expressions/lambda/arrow-expression.bal]: Nope, pkg level variables are anyway visible. Closures are outer scope from your current. So they need to have separate test cases.

Comment (Anoukh) [reply tests/ballerina-unit-test/src/test/resources/test-src/expressions/lambda/arrow-expression.bal]: Ya that's true. Added a test case in 8cf15f5

Comment (Anoukh) [reply tests/ballerina-unit-test/src/test/resources/test-src/expressions/lambda/arrow-expression.bal]: Ya that's true. Added a test case

**Pull request merged**